utils/rowgroup/rowaggregation.cpp

```diff
@@ -720,31 +721,32 @@ void RowAggregation::setJoinRowGroups(vector<RowGroup>* pSmallSideRG, RowGroup*
 // threads on the PM and by multple threads on the UM. It must remain
 // thread safe.
 //------------------------------------------------------------------------------
-void RowAggregation::resetUDAF(RowUDAFFunctionCol* rowUDAF)
+void RowAggregation::resetUDAF(RowUDAFFunctionCol* rowUDAF, uint64_t funcColsIdx)
 {
     // RowAggregation and it's functions need to be re-entrant which means
     // each instance (thread) needs its own copy of the context object.
     // Note: operator=() doesn't copy userData.
-    fRGContext = rowUDAF->fUDAFContext;
+    fRGContextColl[funcColsIdx] = rowUDAF->fUDAFContext;
```

Comment:
@drrtuy This line is causing a PrimProc crash when running the following query:

```SQL
MariaDB [tpch1]> select o_orderstatus, moda(o_custkey) from orders group by o_orderstatus order by o_orderstatus;
```

I don't have a full context on the problem yet, but in RowAggregation::initialize(), the code path does not reach the else block containing `fRGContextColl.resize(fFunctionCols.size());`.